utils/data-processing-initialization/src/main/java/com/github/cafdataprocessing/utilities/initialization/WorkflowRemover.java

```diff
@@ -0,0 +1,113 @@
+     */
+    public static void removeMatchingWorkflows(ProcessingApisProvider apisProvider, String projectId,
+                                               String workflowNameToRemove) throws ApiException {
+        LOGGER.info("Checking for existing processing workflows that should be removed using name: "
+                +workflowNameToRemove);
+        if(workflowNameToRemove==null){
+            LOGGER.info("Workflow name to use in checking existing processing workflows to remove cannot be null. Workflows will not be checked.");
+            return;
+        }
+        WorkflowsApi workflowsApi = apisProvider.getWorkflowsApi();
+        List<ExistingWorkflow> existingWorkflows = new ArrayList<>();
+        {
+            int pageNum = 1;
+            int pageSize = 100;
+            LOGGER.debug("Retrieving existing processing workflows to check their names");
+            while(true){
+                ExistingWorkflows retrieveWorkflowResult = workflowsApi.getWorkflows(projectId, pageNum, pageSize);
+                existingWorkflows.addAll(retrieveWorkflowResult.getWorkflows());
+                if(retrieveWorkflowResult.getTotalHits() <= pageNum*pageSize){
+                    break;
+                }
+                pageNum++;
+            }
+            LOGGER.debug("Retrieved all existing processing workflows.");
+        }
+
+        if(existingWorkflows.isEmpty()){
+            LOGGER.info("There are no existing processing workflows to remove.");
+            return;
+        }
+        ProcessingRulesApi processingRulesApi = apisProvider.getProcessingRulesApi();
+        for(ExistingWorkflow existingWorkflow: existingWorkflows){
+            Long existingWorkflowId = existingWorkflow.getId();
+            if(existingWorkflow.getName().equals(workflowNameToRemove)){
+                LOGGER.debug("Existing processing workflow matches name: "+workflowNameToRemove+", has ID: " +existingWorkflowId+
+                        ". Workflow will be removed.");
+                List<ExistingProcessingRule> processingRulesToRemove = new ArrayList<>();
+                int pageNum = 1;
+                int pageSize = 100;
+                while(true){
+                    ProcessingRules retrieveProcessingRulesResult =
+                            processingRulesApi.getRules(projectId, existingWorkflowId, pageNum, pageSize);
+                    processingRulesToRemove.addAll(retrieveProcessingRulesResult.getRules());
+                    if(retrieveProcessingRulesResult.getTotalHits() <= pageNum*pageSize){
+                        break;
+                    }
+                    pageNum++;
+                }
+                removeProcessingRules(processingRulesApi, projectId, existingWorkflowId, processingRulesToRemove);
+                LOGGER.debug("Removed all processing rules for workflow: "+existingWorkflowId);
+                workflowsApi.deleteWorkflow(projectId, existingWorkflowId);
```

Comment:
Is there no single call that can be made to remove a workflow and its processing rules together - e.g. what if processing rules are removed but workflow removal encounters an error?